### ISSUE TITLE
Refactor path commands

### ISSUE THREAD

**Issue by JordanMartinez** [JordanMartinez/purescript-halogen-svg-elems#8]
To create a path command like `M0 0 L4 2 z', one has to write this, which uses parenthesis unnecessarily in my opinion:
```purescript
SVG.path [ SA.d [ Abs (M 0.0 0.0), Rel (L 4.0 2.0), Z]
```
Moreover, it builds up a syntax tree representing this commands via an ADT before immediately printing those commands as strings. I don't see why this intermediary syntax tree is necessary.

I propose refactoring this code, so that we do not export constructors at all. Rather constructors become functions. Moreover, each of these functions produce the `String` version of the command but hides it in a Newtype that doesn't have its constructor exported, so no one can modify it once created:
```purescript
SVG.path [ SA.d [ m Abs 0.0 0.0, l Rel 4.0 2.0, z]

newtype PathCommand = PathCommand String

m :: D -> Number -> Number -> PathCommand
m ref x_ y_ = PathCommand $ renderCommand <> show x_ <> " " <> show y_
 where 
 renderCommand = case _ of
  Abs -> "M"
  Rel -> "m"
```

**Comment by JamieBallingall** [comment]
Hasn't this been fixed with commit 805e80f9cbb2d357f792745f116e65f65d5dcb46 ?

Can this issue be closed now?

**Comment by JordanMartinez** [comment]
Probably. Thanks for reminding me about this.
